syft/pkg/cataloger/cpe.go

```diff
@@ -546,68 +513,88 @@ func scanByHyphenOrUnderscore(data []byte, atEOF bool) (advance int, token []byt
 	return 0, nil, nil
 }
 
-func addSeparatorVariations(fields *strset.Set) {
-	for _, field := range fields.List() {
+func addDelimiterVariations(fields *cpeFieldCandidateSet) {
```

Comment:
this is not safe by default since candidate is mutated --though not a problem now, could be tricky for someone adding features to this in the future. Let's make a new candidate object that is a copy of the original.

Reply:
updated in d58e60064dedc38c9f8446daf182ff8a1093cd92